colossus/src/main/scala/colossus/protocols/http/HttpRequest.scala

```diff
@@ -116,13 +141,53 @@ case class HttpRequest(head: HttpRequestHead, body: HttpBody) extends Encoder {
       
   }
 
-  def withHeader(key: String, value: String) = copy(head = head.withHeader(key, value))
 }
 
-object HttpRequest {
+trait HttpRequestBuilding[T] {
```

Comment:
This isn't Real Scala without fancy symbolic names. :) (Please, don't do it :) )